Userland/Libraries/LibWeb/Painting/FilterPainting.cpp

```diff
@@ -98,20 +98,21 @@ void apply_filter_list(Gfx::Bitmap& target_bitmap, Layout::Node const& node, Spa
     }
 }
 
-void apply_backdrop_filter(PaintContext& context, Layout::Node const& node, Gfx::FloatRect const& backdrop_rect, BorderRadiiData const& border_radii_data, CSS::BackdropFilter const& backdrop_filter)
+void apply_backdrop_filter(PaintContext& context, Layout::Node const& node, CSSPixelRect const& backdrop_rect, BorderRadiiData const& border_radii_data, CSS::BackdropFilter const& backdrop_filter)
 {
     // This performs the backdrop filter operation: https://drafts.fxtf.org/filter-effects-2/#backdrop-filter-operation
 
-    auto backdrop_region = backdrop_rect.to_rounded<int>();
+    auto backdrop_region = context.rounded_device_rect(backdrop_rect);
 
     // Note: The region bitmap can be smaller than the backdrop_region if it's at the edge of canvas.
-    Gfx::IntRect actual_region {};
+    DevicePixelRect actual_region {};
```

Comment:
This `DevicePixelRect actual_region {}` will always be empty, `actual_region` is passed by reference to `get_region_bitmap()` and filled in by it (as an out parameter). 

This will be why filter painting is broken (as you bail out checking if `actual_region` is empty just below).

Reply:
Ohhhhhh. Thanks!